### ISSUE TITLE
GMMHMM ignores 'params' argument 

### ISSUE THREAD

**Issue by bsash** [hmmlearn/hmmlearn#382]
## I am reporting a bug

GMMHMM fit method is updating even those parameters that it was told not to update through the *params* argument when initializing the object.

In this below example I am setting ```params='st'```, so the ```fit``` method is supposed to not update the *means, weights, covars*. But it is still updating them.

```
from hmmlearn import hmm
import numpy as np

# Parameters of the GMMHMM
n_comps = 2
n_mix = 2
n_features = 1

# Specify the desired parameters for GMM
desired_weights = np.array([[0.6, 0.4], [0.1, 0.9]])
desired_means = np.array([[10, 20], [30, 40]])
desired_covars = np.array([[2, 3], [4, 5]])

# Init the GMM parameters in correct size
weights_prior = np.zeros((n_comps, n_mix))
means_prior = np.zeros((n_comps, n_mix, n_features))
covars_prior = np.zeros((n_comps, n_mix))

# Put the desired parameters into the priors
weights_prior = desired_weights
means_prior[:,:,0] = desired_means
covars_prior = desired_covars

# Initialize the GMMHMM and print the parameters
gmm_hmm = hmm.GMMHMM(n_components=n_comps, n_mix=n_mix, covariance_type='spherical', params='st', init_params='st')
gmm_hmm.means_ = means_prior
gmm_hmm.weights_ = weights_prior
gmm_hmm.covars_ = covars_prior
print('Before fitting...')
print('means')
print(gmm_hmm.means_)
print('weights')
print(gmm_hmm.weights_)
print('covars')
print(gmm_hmm.covars_)

# Generate data to fit
data_to_fit = np.zeros(1000)
data_to_fit[:300] = 5 + np.random.randn(300)*2
data_to_fit[300:500] = 15 + np.random.randn(200)*3
data_to_fit[500:550] = 25 + np.random.randn(50)*4
data_to_fit[550:] = 50 + np.random.randn(450)*5
data_to_fit = data_to_fit.reshape(-1,1)

# Fit the data
gmm_hmm.fit(data_to_fit)
print('After fitting...')
print('means')
print(gmm_hmm.means_)
print('weights')
print(gmm_hmm.weights_)
print('covars')
print(gmm_hmm.covars_)
```
Output
```
Before fitting...
means
[[[10.]
  [20.]]

 [[30.]
  [40.]]]
weights
[[0.6 0.4]
 [0.1 0.9]]
covars
[[2 3]
 [4 5]]
After fitting...
means
[[[ 5.24645016]
  [15.23326713]]

 [[25.47678965]
  [49.82449515]]]
weights
[[0.62731366 0.37268634]
 [0.10356428 0.89643572]]
covars
[[ 5.19990831  6.8443547 ]
 [20.19250998 23.09001226]]
```

**I took a look at the code and the problem seems to be with the ```_do_mstep``` method of the GMMHMM class.** It's not checking the ```self.params``` before maximizing the means, weights and covars.

```
# Supposed to check that self.params contains 'w'
# if 'w' in self.params: # THIS LINE IS MISSING
# Maximizing weights
alphas_minus_one = self.weights_prior - 1
new_weights_numer = stats['post_mix_sum'] + alphas_minus_one
new_weights_denom = (
stats['post_sum'] + np.sum(alphas_minus_one, axis=1)
        )[:, np.newaxis]
new_weights = new_weights_numer / new_weights_denom

# Similar issue when maximizing covars and weights
```

Is my analysis correct? Would be happy to submit a pull request if it is.

**Comment by anntzer** [comment]
This seems reasonable.

**Comment by anntzer** [comment]
fixed by #383, thanks.
